### ISSUE TITLE
network connection error

### ISSUE THREAD

**Issue by promeneur** [librehat/com.librehat.yahooweather#77]
opensuse 42.2
qt 5.6
you say qt must be >= 5.4

i want a localized french app.

i compile wihtout any problem app
i installed it with as recommended with "sudo make install"

i set the woeid to "580778" for Bordeaux,fr
then i wait a long time the i get an error
"erreur 1 vérifiez votre connexion" <=> "error 1 check your connection"

in a VM i installed opensuse 42.2 and i update to qt 5.7
then
no problem!

i don't want to update to qt 5.7 because i don't want to get some trouble with my 42.2

**Comment by smitgd** [comment]
Hi promeneur,
I am running Fedora 23 with Qt Version 5.6.2 and it seems to work fine. I also have another computer with Qt 5.7.1 that, of course, also works. I haven't tried suse 42.2 however.
The error you report sounds like a network connection error timeout. Did you try to connect more than one time before upgrading to Qt 5.7? Maybe it was just an intermittent connection error? Maybe click the refresh button again after the timeout and see if it fails again. Using your woeid on my Qt 5.6.2 works fine from here with no timeout.
-gene
(Just a user and have contributed some to this project.)

**Comment by smitgd** [comment]
I tried Opensuse 42.2 in virtualBox VM and it seems to work OK for your location code and mine. In KDE Info it shows Qt version 5.6.1

**Comment by promeneur** [comment]
let's be clear about opensuse 42.2

- the VM 42.2
i updated to qt 5.7 before having the idea to setup yahoo weather. thus yahoo weather never encountered any problem.
when i say "updated to qt 5.7" this means i updated with the following 42.2 repos:
KDE Extra
KDE Applications
KDE framework 5
QT 5

- my pc with 42.2 
i updated it with the following repos:
KDE Extra
KDE Applications

with my pc i always got a connection problem with yahoo weather. i tried many times.
when i say "qt 5.6" it's accurately 5.6.1
i use many internet apps with not any problem.

i have a proxy: privoxy to avoid commercials. Does yahoo weather use system network settings? in this case it uses privoxy. i assume it may be the cause of problem.
tomorrow i will test without privoxy.

**Comment by smitgd** [comment]
I assume your VM 42.2 is running on your pc with 42.2? If so, it doesn't seem to be affected by privoxy since weather widget works (at least with qt 5.7). I think the weather widget directly on your pc with qt 5.6 would also bypass privoxy since, according to privoxy FAQ, it must be requested to be used in your browser (you enter the ip addr of the privoxy server?). I don't see any "system network setting" in KDE to use a proxy ip. But, I could be wrong. Let me know what you determine.

**Comment by smitgd** [comment]
I went ahead and installed privoxy in Fedora. And, I now do see the KDE setting for proxy. It affects konqueror browser and I assume other KDE apps. If I set KDE to use privoxy proxy (at 127.1.1.0:8118) the weather connection times out as you describe. However, if I put yahooapis.com in the exceptions list it connects and works with no problem. Don't know why weather widget won't connect through privoxy without this exception.
I don't know much about privoxy but there is a "toggle" /etc/privoxy/config parameter that makes it a "content-neutral" proxy by setting it to 0 and restarting privoxy didn't eliminate the need for the exception.
